### PR TITLE
Do a card verification during tokenizing in the Checkout.com gateway

### PR DESCRIPTION
**Pull request by dbackowski**
This will do a card verification during tokenizing in the Checkout.com gateway

### CODE REVIEW

**Comment by dbackowski** [comment]
Screenshot 2021-08-17 at 15 33 10

File: lib/active_merchant/billing/gateways/checkout_v2.rb [hunk 1]
```diff
@@ -178,13 +184,16 @@ def add_customer_data(post, options)
...
+          post[:customer][:name] = options[:billing_address][:name]
+          billing_address = {}
+          billing_address[:billing_address] = {}
+          billing_address[:billing_address][:address_line1] = address[:address1] unless address[:address1].blank?
+          billing_address[:billing_address][:address_line2] = address[:address2] unless address[:address2].blank?
+          billing_address[:billing_address][:city] = address[:city] unless address[:city].blank?
+          billing_address[:billing_address][:state] = address[:state] unless address[:state].blank?
+          billing_address[:billing_address][:country] = address[:country] unless address[:country].blank?
+          billing_address[:billing_address][:zip] = address[:zip] unless address[:zip].blank?
+          post[:source] ? post[:source].merge!(billing_address) : post.merge!(billing_address)
```
Comment (himn1) [inline lib/active_merchant/billing/gateways/checkout_v2.rb:196]: It looks like the `billing_address` is empty {}. Do you mean `billing_address[:billing_address]` ?

**Review by himn1 — COMMENTED** [review]
left 1 comment

File: lib/active_merchant/billing/gateways/checkout_v2.rb (hunk 1 above)
Comment (dbackowski) [reply lib/active_merchant/billing/gateways/checkout_v2.rb:196]: The `billing_address` is set as an empty hash, then we assign all data to it. 
In the end, it is merged in the `post` hash:
```ruby
post[:source] ? post[:source].merge!(billing_address) : post.merge!(billing_address)
```

**Review by himn1 — APPROVED** [review]

**Review by pohor — APPROVED** [review]
re 🚀

**Pull request merged**